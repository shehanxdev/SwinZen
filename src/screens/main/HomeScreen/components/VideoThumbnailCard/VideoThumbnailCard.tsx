import React from 'react';
import { ImageBackground, View } from 'react-native';

import { PlayButtonIcon } from '@sz/components';
import { tw } from '@sz/config';

import { VideoUploadCardFooter } from '../VideoUploadCard';

export function VideoThumbnailCard() {
  return (
    <View style={tw`h-54.5 rounded-2.5 overflow-hidden`}>
      <ImageBackground
        source={{ uri: 'https://i.ibb.co/XFvHx8J/Rectangle-132.png' }}
        resizeMode="cover"
        style={tw`flex-1 items-center justify-center`}>
        <PlayButtonIcon />
      </ImageBackground>
      <VideoUploadCardFooter date="04 JUN 2022 • Faceview" results="10 • Pass" />
    </View>
  );
}
