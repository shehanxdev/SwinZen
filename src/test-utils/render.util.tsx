import { NavigationContext } from '@react-navigation/native';
import { render } from '@testing-library/react-native';
import React, { ReactNode } from 'react';
import Config from 'react-native-config';
import { Provider as PaperProvider } from 'react-native-paper';
import { Provider } from 'react-redux';

import { paperTheme } from '@sz/config';
import { ConfigService } from '@sz/services';
import { store } from '@sz/stores';

type RenderParams = Parameters<typeof render>;

export function renderWithProviders(ui: RenderParams[0], options?: RenderParams[1]) {
  ConfigService.setCallback(key => Config[key]);

  const BaseProviders = ({ children }: { children: ReactNode }) => (
    <Provider store={store}>
      <NavigationContext.Provider
        //@ts-ignore
        value={{
          isFocused: () => true,
          addListener: jest.fn(() => jest.fn()),
        }}>
        <PaperProvider theme={paperTheme}>{children}</PaperProvider>
      </NavigationContext.Provider>
    </Provider>
  );

  return render(ui, { wrapper: BaseProviders, ...options });
}
