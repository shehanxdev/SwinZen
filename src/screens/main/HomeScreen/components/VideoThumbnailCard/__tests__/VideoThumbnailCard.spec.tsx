import * as React from 'react';

import { renderWithProviders } from '@sz/test-utils';

import { VideoThumbnailCard } from '../VideoThumbnailCard';

describe('Home screen', () => {
  const mockVideo = {
    id: 'id4',
    userId: 'string',
    name: 'string',
    videoUrl: 'string',
    videoType: 'Face View',
    thumbnailUrl: 'https://i.ibb.co/XFvHx8J/Rectangle-132.png',
    grading: 4.7,
    createdAt: '2023-06-04T13:22:29.181Z',
  };
  const getRenderedScreen = () => renderWithProviders(<VideoThumbnailCard video={mockVideo} />);

  it(`should render VideoThumbnailCard correctly`, async () => {
    const renderer = getRenderedScreen();
    const renderTree = renderer.toJSON();
    expect(renderTree).toMatchSnapshot();
  });
});
