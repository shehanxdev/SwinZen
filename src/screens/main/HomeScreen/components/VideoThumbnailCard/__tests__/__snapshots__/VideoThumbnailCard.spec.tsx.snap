// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home screen should render VideoThumbnailCard correctly 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "position": "relative",
      }
    }
    testID="VideoThumbnailCardTestID"
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        {
          "borderBottomLeftRadius": 20,
          "borderRadius": 10,
          "height": 218,
          "overflow": "hidden",
        }
      }
    >
      <Image
        resizeMode="cover"
        source={
          {
            "uri": "https://i.ibb.co/XFvHx8J/Rectangle-132.png",
          }
        }
        style={
          [
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            {
              "height": 218,
              "width": undefined,
            },
            undefined,
          ]
        }
      />
      <BVLinearGradient
        colors={
          [
            16144405,
            4294334485,
          ]
        }
        endPoint={
          {
            "x": 0,
            "y": 0,
          }
        }
        locations={
          [
            0.74,
            1,
          ]
        }
        startPoint={
          {
            "x": 1,
            "y": 0,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "display": "flex",
              "height": "100%",
              "justifyContent": "center",
            }
          }
        >
          <SvgMock
            fill="#F8F9FA"
            height={46}
            width={46}
          />
          <View
            style={
              {
                "backgroundColor": "#F65815",
                "borderRadius": 9999,
                "bottom": 0,
                "display": "flex",
                "elevation": 3,
                "height": 40,
                "justifyContent": "center",
                "left": 0,
                "position": "absolute",
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 1,
                  "width": 1,
                },
                "shadowOpacity": 0.125,
                "shadowRadius": 3,
                "width": 40,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#F8F9FA",
                    "lineHeight": 28,
                  },
                  {
                    "fontFamily": "SourceSansPro-SemiBold",
                    "fontSize": 24,
                    "fontWeight": "600",
                    "textAlign": "center",
                  },
                ]
              }
            >
              4.7
            </Text>
          </View>
        </View>
      </BVLinearGradient>
    </View>
    <View>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#F65815",
                "lineHeight": 24,
              },
              {
                "fontFamily": "SourceSansPro-SemiBold",
                "fontSize": 16,
                "fontWeight": "600",
                "textAlign": "center",
              },
            ]
          }
        >
          FAIL
        </Text>
        <Text
          style={
            [
              {
                "color": "#F8F9FA",
                "lineHeight": 24,
              },
              {
                "fontFamily": "SourceSansPro-SemiBold",
                "fontSize": 16,
                "fontWeight": "600",
                "textAlign": "center",
              },
            ]
          }
        >
          04 JUN 2023 â€¢ Face View
        </Text>
      </View>
    </View>
  </View>
</View>
`;
